cylc/flow/network/multi.py

```diff
@@ -220,21 +239,36 @@ def _process_response(
 
 
 def _report(
-    response: dict,
+    response: Union[dict, list],
 ) -> Tuple[Optional[str], Optional[str], bool]:
     """Report the result of a GraphQL operation.
 
     This analyses GraphQL mutation responses to determine the outcome.
 
     Args:
-        response: The GraphQL response.
+        response: The workflow server response (NOT necessarily conforming to
+        GraphQL execution result spec).
 
     Returns:
         (stdout, stderr, outcome)
 
     """
     try:
         ret: List[Tuple[Optional[str], Optional[str], bool]] = []
+        if not isinstance(response, dict):
+            if isinstance(response, list) and response[0].get('error'):
+                # If operating on workflow running in older Cylc version,
+                # may get a error response like [{'error': '...'}]
+                if response[0]['error'].get('message') in KNOWN_INCOMPAT:
+                    raise Exception(
```

Comment:
Is raising an exception here the right thing to do?

We may be operating on multiple workflows at multiple versions. Some of the operations may have succeeded, some may still be being processed.

Reply:
This function is called once per workflow, and the raised Exception is caught down below and used to print the error message.
```console
$ cylc trigger w1//1/foo w2//20250221T0000Z/foo
w1: Command queued
w2: Error processing command:
        Exception: This command is no longer compatible ...
```